### ISSUE TITLE
V3: "0.0.0.0:443: bind: address already in use" when deploying v3 on a server running v2

### ISSUE THREAD

**Issue by farhatahmad** [bigbluebutton/greenlight#4265]
**Describe the bug**
When deploying v3 on a server that is currently running v2 with BBB, you encounter this error when running `docker compose up -d`:

```
"Error response from daemon failed programming external connectivity on endpoint nginx (....): Error starting userland proxy: listen tcp4 [0.0.0.0:443](http://0.0.0.0:443/): bind: address already in use"
```

As we would expect, there's a conflict between the nginx of v3 and the nginx of BigBlueButton + v2.

The fully correct way would be to install v3 on a separate machine, but not everyone has access to unlimited resources

**Comment by hiroshisuga** [comment]
Can a separate GLv3  server govern multiple BBB servers? If so, I dare prepare one!

One problem for setting up a separate v3 server would be the data migration from v2 because you are supposed to run v2 and v3 together... any workaround?

**Comment by debuglevel** [comment]
Maybe see also #4209 as a working relative URL root would be required to run both V2 and V3 on the same machine.

As a temporary workaround to run both on the same machine you could try to run V3 on port 443 and move V2 to port 8443.

**Comment by hiroshisuga** [comment]
Hi @debuglevel , thanks for the comment.
I am even unable to run v3 without the v2 running. After I stop v2 by docker-compose down, I still find the port 443 and 80 are used by nginx of BBB, and thus running v3 fails at launching GL's nginx image (ports already used error). 

I am a bit confused. Am I correct in saying the current explanation at https://docs.bigbluebutton.org/greenlight_v3/gl3-install.html only explain how to install GLv3 in a separate PC without co-running BBB?

**Comment by debuglevel** [comment]
Hi @hiroshisuga,
Oh yes, sorry. I never even considered trying Greenlight3 on a BBB host. This will just not work with `docker-compose.yml` of ["greenlight-run"](https://github.com/bigbluebutton/greenlight-run) as it deploys its own nginx stuff at port 80 and 443 - it would only work on a host with basically nothing running. I guess for now, this is rather something to facilitate development.

At the current state of greenlight3 I would recommend to use a separate machine to try it out.

To be honest: if you really want to invest a bit of time and know a bit about all that system administration stuff, I would recommend to see `docker-compose.yml` rather as a mere idea of how greenlight *could* be run. On my machine, I removed the `nginx` (and replaced it with a reverse-proxy on my machine), `certbot` (and used my own SSL certs in nginx) and `keycloak` (and used the built-in auth backend) containers.

The whole combination of https://github.com/bigbluebutton/greenlight, https://github.com/bigbluebutton/greenlight-run and https://docs.bigbluebutton.org/greenlight_v3/gl3-install.html is just not production-ready for now, and in my opinion not even beta-ready.

**Comment by hiroshisuga** [comment]
Thanks @debuglevel for the clarification. I will try to prepare a separate PC.

**Comment by hiroshisuga** [comment]
I managed to run GLv3 in the same machine that I use to run BBB and GLv2. You need to stop BBB and GLv2, as we would expect.

I write the procedure here just for those who misunderstand the document like I did.
sudo bbb-conf --stop
sudo systemctl stop nginx
(at greenlight directlry)
sudo docker-compose down
Then you can run GLv3 at greenlight-run directory by 
sudo docker compose up -d

No wonder that this GLv3 cannot manage the BBB instances in the same machine. 

It would be kind if the document has a sentence like "running GLv3 together with BBB or GLv2 is not possible right for now".

**Comment by hiroshisuga** [comment]
I've managed to run GLv3 together with GLv2 and BBB2.6, by changing the port used by GLv3 to 8443. I didn't want to change the BBB and GLv2 ports because my people is still using it. Here is what I changed (after setting up GLv3 normally according to the document with BBB and GLv2 turned off):

- nginx: ports: "443:443" -> "8443:8443" in ~/greenlight-run/docker-compose.yml
- add ":8443" in the URLs of OPENID_CONNECT_ISSUER and OPENID_CONNECT_REDIRECT in ~/greenlight-run/data/greenlight/.env
- replace 443 by 8443 at all "listen" variables in ~/greenlight-run/data/nginx/sites.template-docker and ~/greenlight-run/data/nginx/sites.template-local

Then start GLv3, and open Keycloak by opening the URL of kc.mydomain.xxx with :8443 port added.

- go to Clients: greenlight: Credentials: Valid Redirect URIs, and add :8443 in the URL.
- open Microsoft Azure (in case you use Office365 as identity provider) and add a new URL with :8443 at the app registration.

Then restart GLv3. Make sure to put the BBB's endpoint and secret in ~/greenlight-run/data/greenlight/.env . If you open gl.mydomain.xxx:8443, GLv3 will work with GLv2 running in parallel.

One problem is that the recording is not visible on GLv3. I have confirmed the recording itself is done without problems. I've found errors in /var/log/bigbluebutton/post_publishing.log, which seemed to indicate that BBB's publishing script still tried to send the URL without :8443. So obviously we need some change in BBB's side to make the recordings visible. 

I hope this helps somebody who wishes to contribute to the development of GLv3 by installing it on their running BBB and GLv2 server.
